tests/Doctrine/Tests/DBAL/Functional/ModifyLimitQueryTest.php

```diff
@@ -195,4 +200,51 @@ private function assertLimitResult(
             self::assertCount(count($expectedResults), $data);
         }
     }
+
+    public function testLimitWhenOrderByWithSubqueryWithOrderBy(): void
+    {
+        $platform = $this->connection->getDatabasePlatform();
+        if ($platform instanceof DB2Platform) {
+            self::markTestSkipped('Generated SQL is invalid');
+        }
+
+        if (
+            $platform instanceof SQLServer2008Platform
+            && ! $platform instanceof SQLServer2012Platform
+        ) {
+            self::markTestSkipped('SQLServer2008 cannot handle LIMIT in subqueries');
+        }
+
+        $driver = $this->connection->getDriver();
+        if ($driver instanceof PDO\OCI\Driver || $driver instanceof OCI8Driver) {
+            $this->markTestIncomplete(
+                <<<'ERROR'
+ORA-00904: "T1"."ID": invalid identifier
+Doctrine\DBAL\Exception\InvalidFieldNameException: An exception occurred while executing:
+ SELECT a.*
+ FROM (SELECT test_int
+      FROM modify_limit_table2 T1
+      ORDER BY (SELECT a.*
+                FROM (SELECT test_int FROM modify_limit_table2 T2 WHERE T1.id = T2.id ORDER BY test_int ASC) a
+                WHERE ROWNUM <= 1) ASC) a
+ WHERE ROWNUM <= 10
+ERROR
+            );
+        }
+
+        $this->connection->insert('modify_limit_table2', ['test_int' => 3]);
+        $this->connection->insert('modify_limit_table2', ['test_int' => 1]);
+        $this->connection->insert('modify_limit_table2', ['test_int' => 2]);
+
+        $subquery = $this->connection->createQueryBuilder()
+            ->select('test_int')
+            ->from('modify_limit_table2', 'T2')
+            ->where('T1.id=T2.id')
+            ->orderBy('test_int')
+            ->setMaxResults(1);
```

Comment:
This still doesn't reproduce the reported issue. The LIMIT clause is applied to the sub-query which doesn't have a subquery in it. While the issue is about applying a LIMIT clause to a query with a sub-query.

Reply:
In `assertLimitResult()` LIMIT is applied to query and as parameter query with the subquery is passed.

Reply:
Why then do you need to have a LIMIT clause in the sub-query? The query in the unit test doesn't have it.

This is why it doesn't on Oracle: by wrapping it in a sub-query we're making it a grand-child of the top-level query which is unavailable in its context ([ref](https://community.oracle.com/tech/developers/discussion/1040136/simple-subquery-with-error-ora-00904-invalid-identifier)).

Reply:
I limit the subquery otherwise I am getting the error that the subquery returned more than 1 row. I'll change the unit test.

I tested this query for Oracle on [livesql.oracle](https://livesql.oracle.com/) and its working, but Oracle Database 19c is there.

Reply:
By doing so, you're making the test not work on half of the supported platforms. You can limit the number of the rows selected by the inner query by something like `SELECT T1.id`. Again, the test and the fix are about applying a limit to the outer query, not to the limit one.

Maybe we could start with a practical case. Why do you need this?

Reply:
I made a mistake with error that subquery returns more than 1 line. The error occurred before my query change.
In my work I ran into this bug and rewrote the subquery to join, but thought doctrine should handle it.